cmd/node_builder.go

```diff
@@ -301,27 +311,33 @@ func DefaultBaseConfig() *BaseConfig {
 				BandwidthRateLimit:  0,
 				BandwidthBurstLimit: middleware.LargeMsgMaxUnicastMsgSize,
 			},
-			GossipSubRPCValidationConfigs: &p2pbuilder.GossipSubRPCValidationConfigs{
-				NumberOfWorkers: validation.DefaultNumberOfWorkers,
-				GraftLimits: map[string]int{
-					validation.DiscardThresholdMapKey: validation.DefaultGraftDiscardThreshold,
-					validation.SafetyThresholdMapKey:  validation.DefaultGraftSafetyThreshold,
-					validation.RateLimitMapKey:        validation.DefaultGraftRateLimit,
+			GossipSubConfig: p2pbuilder.DefaultGossipSubConfig(),
+			GossipSubRPCInspectorsConfig: &GossipSubRPCInspectorsConfig{
```

Comment:
I recommend introducing a `Validate` method for each configuration object to ensure that all required fields are properly set and not left with the default Golang zero value. This is essential because, as we've seen from previous incidents, unset parameters can go unnoticed and lead to difficult-to-trace issues. Ideally, `NetworkConfig` should be composed of sub-config objects, each having its own `Validate` method. Before accessing the configuration values, `NetworkConfig.Validate()` should be called, invoking the Validate methods of its sub-configs. Each leaf `Validate` method should panic if it encounters any unexpected values. This approach will help us maintain the integrity of the configuration and prevent potential problems down the line.